### PR TITLE
support failed "go test"

### PR DESCRIPTION
**Pull request by haya14busa**
"go test" can creates coverprofile even when some tests fails.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -45,12 +46,27 @@ func usage() {
 	os.Exit(2)
 }
 
+type ExitError struct {
```
Comment (review-dog) [inline main.go:49]: <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported type ExitError should have comment or be unexported

File: main.go [hunk 2]
```diff
@@ -142,23 +168,56 @@ func getPkgs(pkg string) ([]string, error) {
 }
 
 // coverage runs test for the given pkg and returns cover profile.
-func coverage(pkg string, optionalArgs []string, v bool) ([]*cover.Profile, error) {
-	f, err := ioutil.TempFile("", "goverage")
+// success indicates "go test" succeeded or not. coverage may return profiles
+// even when success=false. When "go test" fails, coverage outputs "go test"
+// result to stdout even when verbose=false.
+func coverage(pkg string, optArgs []string, verbose bool) (profiles []*cover.Profile, err error, success bool) {
```
Comment (review-dog) [inline main.go]: <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
error should be the last type when returning multiple items

File: example/fail/sub/sub.go [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package sub
+
+func ok() {
+	_ = "ok"
+}
+
+func CoveredFromRoot() {
```
Comment (review-dog) [inline example/fail/sub/sub.go:7]: <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported function CoveredFromRoot should have comment or be unexported

File: main.go (hunk 2 above)
Comment (haya14busa) [reply main.go]: hmm... that's reasonable here

**Pull request merged**